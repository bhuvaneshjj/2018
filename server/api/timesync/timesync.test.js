'use strict';

const { API_TIMESYNC, apiCall } = require('../../../common/common');
const createWsServer = require('../../createWsServer');
const createWsClient = require('../createWsClient');

const port = 5602;
const wsServer = createWsServer({});
const wsClient = createWsClient({ port });

beforeAll(async () => {
  wsServer.listen(port);
  await wsClient.open();
});

afterAll(async () => {
  await wsClient.close();
  await wsServer.close();
});

it('timesync', async () => {
  const { requestId, ...response } = await wsClient.sendRequest(
    apiCall({
      endpoint: API_TIMESYNC,
      request: { clientTime: new Date('2018-12-01T10:38:08.361Z').toJSON() },
    })
  );
  expect(response.response.serverTime).toBeDefined();
  response.response.serverTime = '2018-12-01T10:38:08:364Z';
  expect(response).toMatchSnapshot();
});
