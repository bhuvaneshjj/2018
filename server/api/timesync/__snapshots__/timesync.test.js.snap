// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timesync 1`] = `
Object {
  "code": "ok",
  "response": Object {
    "clientTime": "2018-12-01T10:38:08.361Z",
    "serverTime": "2018-12-01T10:38:08:364Z",
  },
}
`;
